presto-main/src/main/java/com/facebook/presto/ClientRequestFilterManager.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto;
+
+import com.facebook.presto.spi.ClientRequestFilter;
+import com.facebook.presto.spi.ClientRequestFilterFactory;
+import com.facebook.presto.spi.PrestoException;
+import com.google.common.collect.ImmutableList;
+
+import javax.annotation.concurrent.GuardedBy;
+import javax.annotation.concurrent.ThreadSafe;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.facebook.presto.spi.StandardErrorCode.CONFIGURATION_INVALID;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+@ThreadSafe
+public class ClientRequestFilterManager
+{
+    private List<ClientRequestFilterFactory> factories = new CopyOnWriteArrayList<>();
+    @GuardedBy("this")
+    private volatile List<ClientRequestFilter> filters = ImmutableList.of();
+    private final AtomicBoolean loaded = new AtomicBoolean();
+
+    public void registerClientRequestFilterFactory(ClientRequestFilterFactory factory)
+    {
+        factories.add(factory);
+    }
+
+    public void loadClientRequestFilters()
+    {
+        if (!loaded.compareAndSet(false, true)) {
+            throw new PrestoException(CONFIGURATION_INVALID, "loadClientRequestFilters can only be called once.");
+        }
+
+        filters = factories.stream()
+                .map(factory -> factory.create(factory.getName()))
```

Comment:
This construct is strange.  See above comment.

Reply:
What I meant was, please don't pass in `factory.getName()` into the `create` method, there is no need because it can just internally call `getName()`.